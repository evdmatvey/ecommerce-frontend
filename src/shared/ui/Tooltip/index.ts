export { default } from './ui/Tooltip';
