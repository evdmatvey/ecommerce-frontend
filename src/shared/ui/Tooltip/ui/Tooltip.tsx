import clsx from 'clsx';
import { ReactNode, useState } from 'react';
import styles from './Tooltip.module.css';

interface TooltipProps {
  position?: 'top' | 'right' | 'bottom' | 'left';
  title: string;
  children: ReactNode;
}

const Tooltip = ({ position = 'top', title, children }: TooltipProps) => {
  const [isActive, setActive] = useState(false);

  const showTooltip = () => setActive(true);
  const hideTooltip = () => setActive(false);

  const classes = clsx(styles.title, styles[position]);

  return (
    <div
      className={styles.root}
      onMouseEnter={showTooltip}
      onMouseLeave={hideTooltip}
      data-testid="tooltip-testid"
    >
      {isActive && (
        <div className={classes} data-testid="title-testid">
          {title}
        </div>
      )}
      {children}
    </div>
  );
};

export default Tooltip;
