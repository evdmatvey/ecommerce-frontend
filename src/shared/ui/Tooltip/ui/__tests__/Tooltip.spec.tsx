import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Tooltip from '../Tooltip';

const tooltipTestIds = {
  TOOLTIP: 'tooltip-testid',
  TITLE: 'title-testid',
};

const TEST_TEXT = 'Tooltip!';
const TEST_CONTENT = <span>tooltip</span>;

describe('Tooltip logic', () => {
  it('Should render tooltip correctly', async () => {
    render(<Tooltip title={TEST_TEXT}>{TEST_CONTENT}</Tooltip>);
    const tooltip = screen.getByTestId(tooltipTestIds.TOOLTIP);
    fireEvent.mouseEnter(tooltip);

    const title = screen.queryByTestId(tooltipTestIds.TITLE);

    expect(screen.queryByTestId(tooltipTestIds.TITLE)).toBeInTheDocument();
    expect(title).toHaveTextContent(TEST_TEXT);
    expect(tooltip.getElementsByTagName('span')[0]).toHaveTextContent(
      'tooltip',
    );
  });
});

describe('Tooltip snapshots', () => {
  it('Tooltip', () => {
    const { asFragment } = render(<Tooltip title="Title">{TEST_TEXT}</Tooltip>);
    expect(asFragment()).toMatchSnapshot();
  });
});
