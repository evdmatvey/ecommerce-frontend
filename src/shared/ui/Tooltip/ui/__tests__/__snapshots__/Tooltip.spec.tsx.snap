// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip snapshots > Tooltip 1`] = `
<DocumentFragment>
  <div
    class="_root_727287"
    data-testid="tooltip-testid"
  >
    Tooltip!
  </div>
</DocumentFragment>
`;
